isisd/isis_spf.c

```diff
@@ -1480,7 +1463,62 @@ DEFUN (show_isis_topology,
 
 		vty_out(vty, "\n");
 	}
+}
+
+DEFUN(show_isis_topology, show_isis_topology_cmd,
+      "show " PROTO_NAME
+      " [vrf <NAME|all>] topology"
+#ifndef FABRICD
+      " [<level-1|level-2>]"
+#endif
+      ,
+      SHOW_STR PROTO_HELP VRF_CMD_HELP_STR
+      "All VRFs\n"
+      "IS-IS paths to Intermediate Systems\n"
+#ifndef FABRICD
+      "Paths to all level-1 routers in the area\n"
+      "Paths to all level-2 routers in the domain\n"
+#endif
+)
+{
+	int levels = ISIS_LEVELS;
+	char *vrf_name = NULL;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+	struct listnode *inode, *nnode;
+	struct isis *isis = NULL;
+	int idx = 0;
+
+	if (argv_find(argv, argc, "topology", &idx)) {
+		if (argc < idx + 2)
+			levels = ISIS_LEVEL1 | ISIS_LEVEL2;
+		else if (strmatch(argv[idx + 1]->arg, "level-1"))
+			levels = ISIS_LEVEL1;
+		else
+			levels = ISIS_LEVEL2;
+	}
 
+	if (!im) {
+		vty_out(vty, "IS-IS Routing Process not enabled\n");
+		return CMD_SUCCESS;
+	}
+	ISIS_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
+
+	if (vrf_name) {
```

Comment:
Let's find a way to not cut-n-paste code for the case of a single vrf -vs-  the default vrf

Reply:
Sure, will find a way and do it for the case of a single vrf -vs- the default vrf.